biodatasets/ask_a_patient/ask_a_patient.py

```diff
@@ -0,0 +1,168 @@
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    
+    def _split_generators(self, dl_manager):
+        dl_dir = dl_manager.download_and_extract(_URLs)
+        dataset_dir = os.path.join(dl_dir, "datasets", "AskAPatient")
+        # dataset supports k-folds 
+        splits = []
+        for split_name in [
+            datasets.Split.TRAIN,
+            datasets.Split.VALIDATION,
+            datasets.Split.TEST,
+        ]:
+            for fold_filepath in glob.glob(
+                os.path.join(dataset_dir, f"AskAPatient.fold-*.{split_name}.txt")
+            ):
+                fold_id = re.search("AskAPatient\.fold-(\d)\.", fold_filepath).group(1)
+                document_id = f"{split_name}_{fold_id}"
+                split = datasets.SplitGenerator(
+                    name=f"{split_name}_{fold_id}",
+                    gen_kwargs={
+                        "filepath": fold_filepath,
+                        "document_id": document_id
+                    },
+                )
+                splits.append(split)    
+        return splits
+
+    def _generate_examples(self, filepath, document_id):
+        with open(filepath, "r", encoding="latin-1") as f:
+            for i, line in enumerate(f):
+                id = f"{document_id}_{i}"
+                cui, medical_concept, social_media_text = line.strip().split("\t")
+                if self.config.schema == "source":
+                    yield id, {
+                        "cui": cui,
+                        "medical_concept": medical_concept,
+                        "social_media_text": social_media_text,
+                    }
+                elif self.config.schema == "bigbio_kb":
+                    text_type = "social_media_text"
+                    offset = (0, len(social_media_text))
+                    yield id, {
+                        "id": id,
+                        "document_id": document_id,
+                        "passages": [ 
+                            {
+                                "id": f"{id}_passage",
+                                "type": text_type,
+                                "text": [social_media_text],
+                                "offsets" : [offset],
+                            }
+                        ],
+                        "entities": [ 
+                            {
+                                "id": f"{id}_entity",
+                                "type": text_type,
+                                "text": [social_media_text],
+                                "offsets" : [offset],
+                                "normalized" : [
+                                    {"db_name" : "SNOMED-CT|AMT", "db_id" : cui}
+                                    ]
+                            }
+                        ],
+                        "events": [],
+                        "coreferences": [],
+                        "relations": [],
+                    }
```

Comment:
Can you fix this please? Then IMO it should be good to go. Thank you!

Reply:
It says "no newline at the end of the file"

Reply:
Can you post the stack trace here / on Discord please @sg-wbi? I don't see this when I run on my machine

Reply:
It;s not a bug, it's a "cosmetic" thing. If I go here: https://github.com/bigscience-workshop/biomedical/pull/340/files it shows at bottom of the file

Reply:
Ah! Many thanks for clarifying, fixed now